### PR TITLE
[Doc] Fix broken links in train documentation

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR fixes a few broken links in the train library and documentation.

## Related issue number

Partially addresses #39658.

@matthewdeng @justinvyu Before merging this, the `benchmark_example.rst` is totally orphaned at the moment. It's not part of any example gallery, and there's no real way for people to find it on our docs. What should we do about that?

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/base_trainer.py [hunk 1]
```diff
@@ -571,7 +571,7 @@ def fit(self) -> Result:
             A Result object containing the training result.
 
         Raises:
-            TrainingFailedError: If any failures during the execution of
+            TrainingFailedError\: If any failures during the execution of
```
Comment (woshiyyya) [inline python/ray/train/base_trainer.py:574]: Why are we adding a `\` here?

Comment (peytondmurray) [reply python/ray/train/base_trainer.py:574]: Because it's not valid RST syntax without it.

Comment (peytondmurray) [reply python/ray/train/base_trainer.py:574]: Actually, I take it back: we're using `napoleon` to allow Sphinx to ingest google-style docstrings, so this should be okay. The real culprit is twofold:

1. The line underneath needs to be indented if it's part of the same explanation
2. The exception needs to be fully qualified

Making changes now...

Comment (peytondmurray) [reply python/ray/train/base_trainer.py]: Resolved!

**Comment by matthewdeng** [comment]
@peytondmurray we can go ahead and remove `benchmark_example.rst`! Thanks!

**Comment by peytondmurray** [comment]
Removed. Should be good to merge once docs build+tests pass.

File: doc/source/train/examples/pytorch/torch_data_prefetch_benchmark/benchmark_example.rst [hunk 2]
```diff
@@ -1,53 +0,0 @@
-:orphan:
```
Comment (aslonnie) [inline doc/source/train/examples/pytorch/torch_data_prefetch_benchmark/benchmark_example.rst:1]: can we leave the deletion of this file in a separate PR?

Comment (aslonnie) [reply doc/source/train/examples/pytorch/torch_data_prefetch_benchmark/benchmark_example.rst:1]: ok. saw the other comments.. will just leave it here..

**Review by aslonnie — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]
approved for docs, but @matthewdeng  leaving this PR to you.

**Review by matthewdeng — APPROVED** [review]
ty!

**Pull request merged**